File: Diagrammatize.Design.Pattern/code/python/AbstractFactory/A2/factory/Page.py
import abc

class Page:
    def __init__(self, title, author):
        self.title = title
        self.author = author
        self.content = list()

    def add(self, item):
        self.content.append(item)

    def output(self):
        filename = self.title + ".html"
        writer = open(filename, "w")
        writer.write(self.makeHTML())
        writer.close()
        print(filename + " 编写完成。")

    @abc.abstractmethod
    def makeHTML(self):
        pass

File: Diagrammatize.Design.Patterns/code/cxx/AbstractFactory/Sample/factory/Page.cpp
#include "Page.hpp"
#include <fstream>
#include <iostream>

Page::Page(const std::string& title_, const std::string& author_): title(title_), author(author_) {
}

void Page::add(Item* item) {
    content.push_back(item);
}

void Page::output() {
    std::string filename = title + ".html";
    std::ofstream writer;
    writer.open(filename);
    writer << makeHTML();
    writer.close();
    std::cout << filename << " 编写完成。" << std::endl;
}
